### PR TITLE
fix: use controlled selection components from DataTable for BudgetAssigmentsTable

### PR DESCRIPTION
**Pull request by adamstankiewicz**
# Description

https://2u-internal.atlassian.net/browse/ENT-8048

Ensures "Select all..." functions as expected, with exception of the Paragon bug related to the row count shown in the "Select all..." label (see [this Paragon bug](https://github.com/openedx/paragon/issues/2872) describing the issue with the "Select all..." label's count). Also, does not update the counts shown in the bulk actions or call a "full table selected" specific API endpoint.

https://github.com/openedx/frontend-app-admin-portal/assets/2828721/80cc5599-34db-4545-b982-674089da67ae

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

**Review by brobro10000 — APPROVED** [review]
👍🏽

**Review by iloveagent57 — APPROVED** [review]

**Review by katrinan029 — APPROVED** [review]
LGTM! I'm wondering if the "Select all 25" text should be updated to simply say "Select all". I think it's a bit misleading to indicate that only 25 items are being selected when in reality it could be more or less (i.e. the recording shows all 60 items selected).

**Comment by adamstankiewicz** [comment]
> LGTM! I'm wondering if the "Select all 25" text should be updated to simply say "Select all". I think it's a bit misleading to indicate that only 25 items are being selected when in reality it could be more or less (i.e. the recording shows all 60 items selected).

@katrinan029 Yeah, the misleading "Select all" label is due to a bug upstream with Paragon and is documented in the GitHub issue linked above (https://github.com/openedx/paragon/issues/2872). Given this repo is blocked on upgrading to the latest versions of Paragon due to the React/React Router version upgrades, we'd have no control over the "Select all" label unless we opt to replicate and modify the default `DataTable.ControlledSelectionStatus` the table relies on (personally, I would avoid this approach in favor of waiting for the bug to be fixed upstream separately, once we can upgrade Paragon in the repo).

**Pull request merged**
